source/cgal/intersections.cc

```diff
@@ -359,400 +355,449 @@ namespace CGALWrappers
-  std::vector<std::array<Point<3>, 4>>
-  compute_intersection_of_cells<3, 3, 3, 8, 8>(
-    const std::array<Point<3>, 8> &vertices0,
-    const std::array<Point<3>, 8> &vertices1,
 
-    const double tol)
+      Surface_mesh surf0, surf1, sm;
+      // Subdivide hex into tetrahedrons
+      std::vector<std::array<Point<3>, 4>> vertices;
+      Triangulation3_inexact               tria0, tria1;
+
+      tria0.insert(pts_hex0.begin(), pts_hex0.end());
+      tria1.insert(pts_hex1.begin(), pts_hex1.end());
+
+      for (const auto &c0 : tria0.finite_cell_handles())
+        {
+          const auto &                 tet0 = tria1.tetrahedron(c0);
+          [[maybe_unused]] const auto &tetg0 =
+            CGAL::make_tetrahedron(tet0.vertex(0),
+                                   tet0.vertex(1),
+                                   tet0.vertex(2),
+                                   tet0.vertex(3),
+                                   surf0);
+          for (const auto &c1 : tria1.finite_cell_handles())
+            {
+              const auto &                 tet1 = tria1.tetrahedron(c1);
+              [[maybe_unused]] const auto &tetg1 =
+                CGAL::make_tetrahedron(tet1.vertex(0),
+                                       tet1.vertex(1),
+                                       tet1.vertex(2),
+                                       tet1.vertex(3),
+                                       surf1);
+              namespace PMP = CGAL::Polygon_mesh_processing;
+              const bool test_intersection =
+                PMP::corefine_and_compute_intersection(surf0, surf1, sm);
+              if (PMP::volume(sm) > tol && test_intersection)
+                {
+                  // Collect tetrahedrons
+                  Triangulation3_inexact tria;
+                  tria.insert(sm.points().begin(), sm.points().end());
+                  for (const auto &c : tria.finite_cell_handles())
+                    {
+                      const auto &tet = tria.tetrahedron(c);
+                      vertices.push_back(
+                        {{CGALWrappers::cgal_point_to_dealii_point<3>(
+                            tet.vertex(0)),
+                          CGALWrappers::cgal_point_to_dealii_point<3>(
+                            tet.vertex(1)),
+                          CGALWrappers::cgal_point_to_dealii_point<3>(
+                            tet.vertex(2)),
+                          CGALWrappers::cgal_point_to_dealii_point<3>(
+                            tet.vertex(3))}});
+                    }
+                }
+              surf1.clear();
+              sm.clear();
+            }
+          surf0.clear();
+        }
+      return vertices;
+    }
+
+  } // namespace internal
+
+
+  template <int dim0, int dim1, int spacedim>
+  std::vector<std::array<Point<spacedim>, dim1 + 1>>
+  compute_intersection_of_cells(const ArrayView<Point<spacedim>> &vertices0,
+                                const ArrayView<Point<spacedim>> &vertices1,
```

Comment:
assertion

Reply:
This function will call, depending on the runtime length of the vertices, the previous functions computing intersections (where all the assertions have been added). But I don't see which assumption I should make on the number of vertices for this particular function. I could require that both vertices have size>0, but I'm not sure if you're referring to that.

Reply:
I see now. I just marked this place up like all of the others where an `ArrayView` is passed, assuming that there is an assumed-size for these arrays. Ignore the comment here then.